Referencing `public` `const` from other modules triggers C++ compiler error

We can reference a `global` from another module if it is declared `public`. The docs suggest that a `const` should behave similarly, but this leads to a C++ compiler error.


```.ruby
# a.spicy
module a;

public global x: uint32 = 64;
public const y: uint32 = 64;
```

```.ruby
# b.spicy
module b;

import a;
print(a::x); # Works on its own.
print(a::y); # Fails during C++ compile.
```

```.console
$ spicyc -j b.spicy
/private/var/folders/ht/vtb0rkbd4ws15vx02xdb84zw0000gn/T/hilti-rt-test-IUGmp0/b.cc:29:25: error: no member named 'y' in namespace '__hlt::a'; did you mean simply 'y'?
    ::hilti::rt::print((__hlt::a::y), ::hilti::rt::Bool(true));
                        ^~~~~~~~~~~
                        y
/private/var/folders/ht/vtb0rkbd4ws15vx02xdb84zw0000gn/T/hilti-rt-test-IUGmp0/b.cc:20:48: note: 'y' declared here
    const ::hilti::rt::integer::safe<uint32_t> y = ::hilti::rt::integer::safe<std::uint32_t>{64u};
                                               ^
1 error generated.
[error] spicyc: JIT compilation failed
```

I would have expected a `const` to behave identical to a `global` here.

--- a/hilti/toolchain/src/compiler/codegen/codegen.cc
+++ b/hilti/toolchain/src/compiler/codegen/codegen.cc
@@ -38,7 +38,7 @@ struct GlobalsVisitor : hilti::visitor::PreOrder<void, GlobalsVisitor> {
     bool include_implementation;
 
     std::vector<cxx::declaration::Global> globals;
-    std::vector<cxx::declaration::Global> constants;
+    std::vector<cxx::declaration::Constant> constants;
 
     static void addDeclarations(CodeGen* cg, const Node& module, const ID& module_id, cxx::Unit* unit,
                                 bool include_implementation) {
@@ -135,10 +135,9 @@ struct GlobalsVisitor : hilti::visitor::PreOrder<void, GlobalsVisitor> {
     }
 
     void operator()(const declaration::Constant& n, position_t p) {
-        auto x = cxx::declaration::Global{.id = {cg->unit()->cxxNamespace(), n.id()},
-                                          .type = cg->compile(n.type(), codegen::TypeUsage::Storage),
-                                          .init = cg->compile(n.value()),
-                                          .linkage = "const"};
+        auto x = cxx::declaration::Constant{.id = {cg->unit()->cxxNamespace(), n.id()},
+                                            .type = cg->compile(n.type(), codegen::TypeUsage::Storage),
+                                            .init = cg->compile(n.value())};
 
         constants.push_back(x);
     }

--- a/tests/hilti/codegen/imported-constants.hlt
+++ b/tests/hilti/codegen/imported-constants.hlt
@@ -1,0 +1,16 @@
+# @TEST-EXEC: hiltic -j %INPUT
+# @TEST-DOC: Check usage imported constants. Regression test for #1079.
+
+module B {
+
+import A;
+
+assert A::x == 64;
+
+}
+
+@TEST-START-FILE a.hlt
+module A {
+public const uint<32> x = 64;
+}
+@TEST-END-FILE

